File: Data Structures & Algorithms/non-cyclical-number/submission-6.py
class Solution:
    def isHappy(self, n: int) -> bool:
        seen = set()
        count = 0

        while True:
            count = 0

            while n:
                dig = n % 10
                count += (dig)**2
                n //= 10
            
            if count == 1:
                return True
            elif count in seen:
                return False

            seen.add(count)
            n = count

File: Data Structures & Algorithms/longest-substring-without-duplicates/submission-6.cpp
class Solution {
public:
    int lengthOfLongestSubstring(string s) {
        int l = 0;
        int res = 0;
        unordered_set<char> chars;

        for (int r = 0; r < s.size(); r++) {
            while (chars.count(s[r])) {
                chars.erase(s[l]);
                l++;
            }
            chars.insert(s[r]);
            res = max(res, r - l + 1);
        }

        return res;
    }
};
